src/drivers/indexeddb.js

```diff
@@ -181,27 +182,164 @@
             }
         }
 
+        // Initialize a singleton container for all running localForages.
+        if (!dbContexts) {
+            dbContexts = {};
+        }
+
+        // Get the current context of the database;
+        var dbContext = dbContexts[dbInfo.name];
+
+        // ...or create a new context.
+        if (!dbContext) {
+            dbContext = {
+                // Running localForages sharing a database.
+                forages: [],
+                // Shared database.
+                db: null
+            };
+            // Register the new context in the global container.
+            dbContexts[dbInfo.name] = dbContext;
+        }
+
+        // Register itself as a running localForage in the current context.
+        dbContext.forages.push(this);
+
+        // Create an array of readiness of the related localForages.
+        var ready = [];
+        for (var j = 0; j < dbContext.forages.length; j++) {
+            var forage = dbContext.forages[j];
+            if (forage !== this) { // Don't wait for itself...
+                ready.push(forage.ready());
+            }
+        }
+
+        // Take a snapshot of the related localForages.
+        var forages = dbContext.forages.slice(0);
+
+        // Initialize the connection process only when
+        // all the related localForages are ready.
+        return Promise.all(ready).then(function() {
+            dbInfo.db = dbContext.db;
+            // Get the connection or open a new one without upgrade.
+            return _getOriginalConnection(dbInfo);
+        }).then(function(db) {
+            dbInfo.db = db;
+            if (_isUpgradeNeeded(dbInfo, self._defaultConfig.version)) {
+                // Reopen the database for upgrading.
+                return _getUpgradedConnection(dbInfo);
+            }
+            return db;
+        }).then(function(db) {
+            dbInfo.db = dbContext.db = db;
+            self._dbInfo = dbInfo;
+            // Share the final connection amongst related localForages.
+            for (var k in forages) {
+                var forage = forages[k];
+                if (forage !== self) { // Self is already up-to-date.
+                    forage._dbInfo.db = dbInfo.db;
+                    forage._dbInfo.version = dbInfo.db;
+                }
+            }
+        });
```

Comment:
You are right about `p2.catch().then(function(){ console.log('asdf'); })` not working
I never meant that a `p2.catch()` would alter the promise stored in variable `p2` (if not reassigned).

The fact is that `.catch()` will always return a new Promise so `p2 != p2.catch()`.

The part that confused me, was that it will not change the returned promise status, unless a function is passed as parameter (**any function!**).

In other words `Promise.reject('error').catch(function(){})` will always be resolved.
[Here is a jsbin with the above](https://jsbin.com/cofipapozi/edit?js,console)!?!?

*PS: That's a great discussion for a single PR!*